avni-server-api/src/main/java/org/avni/web/api/GeneralEncounterApiController.java

```diff
@@ -113,29 +125,35 @@ public ResponseEntity put(@PathVariable String id, @RequestBody ApiEncounterRequ
     }
 
     private void updateEncounter(Encounter encounter, ApiEncounterRequest request) throws ValidationException {
-        Individual individual = individualRepository.findByUuid(request.getSubjectId());
-        if (individual == null) {
-            throw new IllegalArgumentException(String.format("Individual not found with UUID '%s'", request.getSubjectId()));
-        }
-
         EncounterType encounterType = encounterTypeRepository.findByName(request.getEncounterType());
         if (encounterType == null) {
             throw new IllegalArgumentException(String.format("Encounter type not found with name '%s'", request.getEncounterType()));
         }
-
+        if (StringUtils.hasLength(request.getExternalId()) && !StringUtils.hasLength(encounter.getLegacyId())) {
```

Comment:
I think this check is not required. We can let callers remove legacy id, if they want to remove it.

Reply:
Okay, removed the check